### ISSUE TITLE
confusion about test

### ISSUE THREAD

**Issue by moonmengmeng** [KaiLv69/UDR#7]
in test_iter2.sh

      python DPR/dense_retriever.py \
           model_file="${ckpt}" \
           qa_dataset=qa_epr \
           ctx_datatsets=[dpr_epr] \
           datasets.qa_epr.dataset_split=$split \
           encoded_ctx_files=["${generate_dense_embeddings_py_output_path}_*"] \
           out_file="${dr_out}" \
           ctx_sources.dpr_epr.setup_type=qa \
           ctx_sources.dpr_epr.task_name=$dataset \
           datasets.qa_epr.task_name=$dataset \
           hydra.run.dir="$PWD/exps/$exp_name/logs"

I'm confusion about ctx_sources.dpr_epr.setup_type=qa, I think in the test set there is no answer. Why it is qa, and not q, where is the answer from?

**Comment by moonmengmeng** [comment]
In the papaer it is Z∗ = top-Kz∈D sim(x, z), it seems like similarity about the input and not QA

**Comment by KaiLv69** [comment]
Hi,

Here, "qa" means that the embeddings of the training set are generated using QA, while the queries in the test set only use the input.
